Errors on GRPC PHP - pecl install grpc

As of today, I am getting this error [on travis](https://travis-ci.org/GoogleCloudPlatform/php-docs-samples/jobs/261942015#L1140) for the command `pecl install grpc`: 

```
$ travis_retry pecl install grpc
# ...
/tmp/pear/temp/grpc/src/php/ext/grpc/channel.c: At top level:
cc1: error: unrecognized command line option "-Wno-parentheses-equality" [-Werror]
cc1: all warnings being treated as errors
make: *** [src/php/ext/grpc/channel.lo] Error 1
ERROR: `make' failed
```

This error seems to be related to https://github.com/grpc/grpc/issues/7571, which was closed almost a year ago.

I only receive this error on PHP 5.6, but I am receiving a different error when running grpc tests on PHP 7.0. The build [times out after 10 minutes](https://travis-ci.org/GoogleCloudPlatform/php-docs-samples/jobs/261994466#L5824), and the logs reveal the following error:

```
E0807 19:31:09.399814488    6759 sync_posix.c:60]            assertion failed: pthread_mutex_lock(mu) == 0
6759 Aborted                 (core dumped) vendor/bin/phpunit

[secure]_time:end:16586363:start=1502133989762474252,finish=1502134269616946002,duration=279854471750
```

Here is the full command output for `pecl install grpc`

```
_retry pecl install grpc
downloading grpc-1.4.3.tgz ...
Starting to download grpc-1.4.3.tgz (2,301,057 bytes)
..................done: 2,301,057 bytes
1048 source files, building
running: phpize
Configuring for:
PHP Api Version:         20131106
Zend Module Api No:      20131226
Zend Extension Api No:   220131226
building in /tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3
running: /tmp/pear/temp/grpc/configure --with-php-config=/home/[secure]/.phpenv/versions/5.6.24/bin/php-config
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for a sed that does not truncate output... /bin/sed
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking whether cc understands -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for PHP prefix... /home/[secure]/.phpenv/versions/5.6.24
checking for PHP includes... -I/home/[secure]/.phpenv/versions/5.6.24/include/php -I/home/[secure]/.phpenv/versions/5.6.24/include/php/main -I/home/[secure]/.phpenv/versions/5.6.24/include/php/TSRM -I/home/[secure]/.phpenv/versions/5.6.24/include/php/Zend -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext/date/lib
checking for PHP extension directory... /home/[secure]/.phpenv/versions/5.6.24/lib/php/extensions/no-debug-zts-20131226
checking for PHP installed headers prefix... /home/[secure]/.phpenv/versions/5.6.24/include/php
checking if debug is enabled... no
checking if zts is enabled... no
checking for re2c... re2c
checking for re2c version... 0.13.5 (ok)
checking for gawk... gawk
checking whether to enable grpc support... yes, shared
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking how to run the C++ preprocessor... g++ -E
checking for ld used by cc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for /usr/bin/ld option to reload object files... -r
checking for BSD-compatible nm... /usr/bin/nm -B
checking whether ln -s works... yes
checking how to recognize dependent libraries... pass_all
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking how to run the C++ preprocessor... g++ -E
checking the maximum length of command line arguments... 1572864
checking command to parse /usr/bin/nm -B output from cc object... ok
checking for objdir... .libs
checking for ar... ar
checking for ranlib... ranlib
checking for strip... strip
checking if cc supports -fno-rtti -fno-exceptions... no
checking for cc option to produce PIC... -fPIC
checking if cc PIC flag -fPIC works... yes
checking if cc static flag -static works... yes
checking if cc supports -c -o file.o... yes
checking whether the cc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no

creating libtool
appending configuration tag "CXX" to libtool
checking for ld used by g++... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC
checking if g++ PIC flag -fPIC works... yes
checking if g++ static flag -static works... yes
checking if g++ supports -c -o file.o... yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... GNU/Linux ld.so
(cached) (cached) checking how to hardcode library paths into programs... immediate
configure: creating ./config.status
config.status: creating config.h
running: make
/bin/bash /tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/libtool --mode=compile cc -fvisibility=hidden     -DOPENSSL_NO_ASM -D_GNU_SOURCE -DWIN32_LEAN_AND_MEAN     -D_HAS_EXCEPTIONS=0 -DNOMINMAX -DGRPC_ARES=0 -I. -I/tmp/pear/temp/grpc -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/include -I/tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/main -I/tmp/pear/temp/grpc -I/home/[secure]/.phpenv/versions/5.6.24/include/php -I/home/[secure]/.phpenv/versions/5.6.24/include/php/main -I/home/[secure]/.phpenv/versions/5.6.24/include/php/TSRM -I/home/[secure]/.phpenv/versions/5.6.24/include/php/Zend -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext/date/lib -I/tmp/pear/temp/grpc/include -I/tmp/pear/temp/grpc/src/php/ext/grpc -I/tmp/pear/temp/grpc/third_party/boringssl/include  -DHAVE_CONFIG_H  -Wall -Werror -Wno-parentheses-equality -Wno-unused-value -std=c11   -c /tmp/pear/temp/grpc/src/php/ext/grpc/byte_buffer.c -o src/php/ext/grpc/byte_buffer
.lo
mkdir src/php/ext/grpc/.libs
 cc -fvisibility=hidden -DOPENSSL_NO_ASM -D_GNU_SOURCE -DWIN32_LEAN_AND_MEAN -D_HAS_EXCEPTIONS=0 -DNOMINMAX -DGRPC_ARES=0 -I. -I/tmp/pear/temp/grpc -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/include -I/tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/main -I/tmp/pear/temp/grpc -I/home/[secure]/.phpenv/versions/5.6.24/include/php -I/home/[secure]/.phpenv/versions/5.6.24/include/php/main -I/home/[secure]/.phpenv/versions/5.6.24/include/php/TSRM -I/home/[secure]/.phpenv/versions/5.6.24/include/php/Zend -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext/date/lib -I/tmp/pear/temp/grpc/include -I/tmp/pear/temp/grpc/src/php/ext/grpc -I/tmp/pear/temp/grpc/third_party/boringssl/include -DHAVE_CONFIG_H -Wall -Werror -Wno-parentheses-equality -Wno-unused-value -std=c11 -c /tmp/pear/temp/grpc/src/php/ext/grpc/byte_buffer.c  -fPIC -DPIC -o src/php/ext/grpc/.libs/byte_buffer.o
/bin/bash /tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/libtool --mode=compile cc -fvisibility=hidden     -DOPENSSL_NO_ASM -D_GNU_SOURCE -DWIN32_LEAN_AND_MEAN     -D_HAS_EXCEPTIONS=0 -DNOMINMAX -DGRPC_ARES=0 -I. -I/tmp/pear/temp/grpc -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/include -I/tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/main -I/tmp/pear/temp/grpc -I/home/[secure]/.phpenv/versions/5.6.24/include/php -I/home/[secure]/.phpenv/versions/5.6.24/include/php/main -I/home/[secure]/.phpenv/versions/5.6.24/include/php/TSRM -I/home/[secure]/.phpenv/versions/5.6.24/include/php/Zend -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext/date/lib -I/tmp/pear/temp/grpc/include -I/tmp/pear/temp/grpc/src/php/ext/grpc -I/tmp/pear/temp/grpc/third_party/boringssl/include  -DHAVE_CONFIG_H  -Wall -Werror -Wno-parentheses-equality -Wno-unused-value -std=c11   -c /tmp/pear/temp/grpc/src/php/ext/grpc/call.c -o src/php/ext/grpc/call.lo
 cc -fvisibility=hidden -DOPENSSL_NO_ASM -D_GNU_SOURCE -DWIN32_LEAN_AND_MEAN -D_HAS_EXCEPTIONS=0 -DNOMINMAX -DGRPC_ARES=0 -I. -I/tmp/pear/temp/grpc -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/include -I/tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/main -I/tmp/pear/temp/grpc -I/home/[secure]/.phpenv/versions/5.6.24/include/php -I/home/[secure]/.phpenv/versions/5.6.24/include/php/main -I/home/[secure]/.phpenv/versions/5.6.24/include/php/TSRM -I/home/[secure]/.phpenv/versions/5.6.24/include/php/Zend -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext/date/lib -I/tmp/pear/temp/grpc/include -I/tmp/pear/temp/grpc/src/php/ext/grpc -I/tmp/pear/temp/grpc/third_party/boringssl/include -DHAVE_CONFIG_H -Wall -Werror -Wno-parentheses-equality -Wno-unused-value -std=c11 -c /tmp/pear/temp/grpc/src/php/ext/grpc/call.c  -fPIC -DPIC -o src/php/ext/grpc/.libs/call.o
/bin/bash /tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/libtool --mode=compile cc -fvisibility=hidden     -DOPENSSL_NO_ASM -D_GNU_SOURCE -DWIN32_LEAN_AND_MEAN     -D_HAS_EXCEPTIONS=0 -DNOMINMAX -DGRPC_ARES=0 -I. -I/tmp/pear/temp/grpc -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/include -I/tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/main -I/tmp/pear/temp/grpc -I/home/[secure]/.phpenv/versions/5.6.24/include/php -I/home/[secure]/.phpenv/versions/5.6.24/include/php/main -I/home/[secure]/.phpenv/versions/5.6.24/include/php/TSRM -I/home/[secure]/.phpenv/versions/5.6.24/include/php/Zend -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext/date/lib -I/tmp/pear/temp/grpc/include -I/tmp/pear/temp/grpc/src/php/ext/grpc -I/tmp/pear/temp/grpc/third_party/boringssl/include  -DHAVE_CONFIG_H  -Wall -Werror -Wno-parentheses-equality -Wno-unused-value -std=c11   -c /tmp/pear/temp/grpc/src/php/ext/grpc/call_credentials.c -o src/php/ext/grpc/call_c
redentials.lo
 cc -fvisibility=hidden -DOPENSSL_NO_ASM -D_GNU_SOURCE -DWIN32_LEAN_AND_MEAN -D_HAS_EXCEPTIONS=0 -DNOMINMAX -DGRPC_ARES=0 -I. -I/tmp/pear/temp/grpc -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/include -I/tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/main -I/tmp/pear/temp/grpc -I/home/[secure]/.phpenv/versions/5.6.24/include/php -I/home/[secure]/.phpenv/versions/5.6.24/include/php/main -I/home/[secure]/.phpenv/versions/5.6.24/include/php/TSRM -I/home/[secure]/.phpenv/versions/5.6.24/include/php/Zend -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext/date/lib -I/tmp/pear/temp/grpc/include -I/tmp/pear/temp/grpc/src/php/ext/grpc -I/tmp/pear/temp/grpc/third_party/boringssl/include -DHAVE_CONFIG_H -Wall -Werror -Wno-parentheses-equality -Wno-unused-value -std=c11 -c /tmp/pear/temp/grpc/src/php/ext/grpc/call_credentials.c  -fPIC -DPIC -o src/php/ext/grpc/.libs/call_credentials.o
/bin/bash /tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/libtool --mode=compile cc -fvisibility=hidden     -DOPENSSL_NO_ASM -D_GNU_SOURCE -DWIN32_LEAN_AND_MEAN     -D_HAS_EXCEPTIONS=0 -DNOMINMAX -DGRPC_ARES=0 -I. -I/tmp/pear/temp/grpc -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/include -I/tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/main -I/tmp/pear/temp/grpc -I/home/[secure]/.phpenv/versions/5.6.24/include/php -I/home/[secure]/.phpenv/versions/5.6.24/include/php/main -I/home/[secure]/.phpenv/versions/5.6.24/include/php/TSRM -I/home/[secure]/.phpenv/versions/5.6.24/include/php/Zend -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext/date/lib -I/tmp/pear/temp/grpc/include -I/tmp/pear/temp/grpc/src/php/ext/grpc -I/tmp/pear/temp/grpc/third_party/boringssl/include  -DHAVE_CONFIG_H  -Wall -Werror -Wno-parentheses-equality -Wno-unused-value -std=c11   -c /tmp/pear/temp/grpc/src/php/ext/grpc/channel.c -o src/php/ext/grpc/channel.lo
 cc -fvisibility=hidden -DOPENSSL_NO_ASM -D_GNU_SOURCE -DWIN32_LEAN_AND_MEAN -D_HAS_EXCEPTIONS=0 -DNOMINMAX -DGRPC_ARES=0 -I. -I/tmp/pear/temp/grpc -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/include -I/tmp/pear/temp/pear-build-[secure]Vzvqp1/grpc-1.4.3/main -I/tmp/pear/temp/grpc -I/home/[secure]/.phpenv/versions/5.6.24/include/php -I/home/[secure]/.phpenv/versions/5.6.24/include/php/main -I/home/[secure]/.phpenv/versions/5.6.24/include/php/TSRM -I/home/[secure]/.phpenv/versions/5.6.24/include/php/Zend -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext -I/home/[secure]/.phpenv/versions/5.6.24/include/php/ext/date/lib -I/tmp/pear/temp/grpc/include -I/tmp/pear/temp/grpc/src/php/ext/grpc -I/tmp/pear/temp/grpc/third_party/boringssl/include -DHAVE_CONFIG_H -Wall -Werror -Wno-parentheses-equality -Wno-unused-value -std=c11 -c /tmp/pear/temp/grpc/src/php/ext/grpc/channel.c  -fPIC -DPIC -o src/php/ext/grpc/.libs/channel.o
In file included from /home/[secure]/.phpenv/versions/5.6.24/include/php/Zend/zend.h:286:0,
                 from /home/[secure]/.phpenv/versions/5.6.24/include/php/main/php.h:35,
                 from /tmp/pear/temp/grpc/src/php/ext/grpc/channel.h:41,
                 from /tmp/pear/temp/grpc/src/php/ext/grpc/channel.c:34:
/tmp/pear/temp/grpc/src/php/ext/grpc/channel.c: In function â€˜create_and_add_channel_to_persistent_listâ€™:
/home/[secure]/.phpenv/versions/5.6.24/include/php/Zend/../TSRM/TSRM.h:164:57: error: â€˜tsrm_lsâ€™ undeclared (first use in this function)
 #define TSRMG(id, type, element) (((type) (*((void ***) tsrm_ls))[TSRM_UNSHUFFLE_RSRC_ID(id)])->element)
                                                         ^
/home/[secure]/.phpenv/versions/5.6.24/include/php/Zend/zend_hash.h:110:28: note: in definition of macro â€˜zend_hash_updateâ€™
   _zend_hash_add_or_update(ht, arKey, nKeyLength, pData, nDataSize, pDest, HASH_UPDATE ZEND_FILE_LINE_CC)
                            ^
/tmp/pear/temp/grpc/src/php/ext/grpc/channel.c:188:3: note: in expansion of macro â€˜PHP_GRPC_PERSISTENT_LIST_UPDATEâ€™
   PHP_GRPC_PERSISTENT_LIST_UPDATE(&EG(persistent_list), key, key_len,
   ^
/home/[secure]/.phpenv/versions/5.6.24/include/php/Zend/zend_globals_macros.h:45:16: note: in expansion of macro â€˜TSRMGâ€™
 # define EG(v) TSRMG(executor_globals_id, zend_executor_globals *, v)
                ^
/tmp/pear/temp/grpc/src/php/ext/grpc/channel.c:188:36: note: in expansion of macro â€˜EGâ€™
   PHP_GRPC_PERSISTENT_LIST_UPDATE(&EG(persistent_list), key, key_len,
                                    ^
/home/[secure]/.phpenv/versions/5.6.24/include/php/Zend/../TSRM/TSRM.h:164:57: note: each undeclared identifier is reported only once for each function it appears in
 #define TSRMG(id, type, element) (((type) (*((void ***) tsrm_ls))[TSRM_UNSHUFFLE_RSRC_ID(id)])->element)
                                                         ^
/home/[secure]/.phpenv/versions/5.6.24/include/php/Zend/zend_hash.h:110:28: note: in definition of macro â€˜zend_hash_updateâ€™
   _zend_hash_add_or_update(ht, arKey, nKeyLength, pData, nDataSize, pDest, HASH_UPDATE ZEND_FILE_LINE_CC)
                            ^
/tmp/pear/temp/grpc/src/php/ext/grpc/channel.c:188:3: note: in expansion of macro â€˜PHP_GRPC_PERSISTENT_LIST_UPDATEâ€™
   PHP_GRPC_PERSISTENT_LIST_UPDATE(&EG(persistent_list), key, key_len,
   ^
/home/[secure]/.phpenv/versions/5.6.24/include/php/Zend/zend_globals_macros.h:45:16: note: in expansion of macro â€˜TSRMGâ€™
 # define EG(v) TSRMG(executor_globals_id, zend_executor_globals *, v)
                ^
/tmp/pear/temp/grpc/src/php/ext/grpc/channel.c:188:36: note: in expansion of macro â€˜EGâ€™
   PHP_GRPC_PERSISTENT_LIST_UPDATE(&EG(persistent_list), key, key_len,
                                    ^
/tmp/pear/temp/grpc/src/php/ext/grpc/channel.c: At top level:
cc1: error: unrecognized command line option "-Wno-parentheses-equality" [-Werror]
cc1: all warnings being treated as errors
make: *** [src/php/ext/grpc/channel.lo] Error 1
ERROR: `make' failed
```

--- a/package.xml
+++ b/package.xml
@@ -22,8 +22,9 @@
  </stability>
  <license>BSD</license>
  <notes>
-- Fixed hang bug when fork() was used #11814
-- Channel are now by default persistent #11878
+- Fixed a Windows installation issue #12108
+- Fixed a MacOS mutex segfault #12109
+- Fixed a ZTS compilation issue #12109
  </notes>
  <contents>
   <dir baseinstalldir="/" name="/">

--- a/src/php/ext/grpc/channel.c
+++ b/src/php/ext/grpc/channel.c
@@ -173,7 +173,7 @@ void create_and_add_channel_to_persistent_list(
     grpc_channel_args args,
     wrapped_grpc_channel_credentials *creds,
     char *key,
-    php_grpc_int key_len) {
+    php_grpc_int key_len TSRMLS_DC) {
   php_grpc_zend_resource new_rsrc;
   channel_persistent_le_t *le;
   // this links each persistent list entry to a destructor
@@ -308,7 +308,7 @@ PHP_METHOD(Channel, __construct) {
   } else if (!(PHP_GRPC_PERSISTENT_LIST_FIND(&EG(persistent_list), key,
                                              key_len, rsrc))) {
     create_and_add_channel_to_persistent_list(
-        channel, target, args, creds, key, key_len);
+        channel, target, args, creds, key, key_len TSRMLS_CC);
   } else {
     // Found a previously stored channel in the persistent list
     channel_persistent_le_t *le = (channel_persistent_le_t *)rsrc->ptr;
@@ -318,7 +318,7 @@ PHP_METHOD(Channel, __construct) {
          strcmp(creds->hashstr, le->channel->creds_hashstr) != 0)) {
       // somehow hash collision
       create_and_add_channel_to_persistent_list(
-          channel, target, args, creds, key, key_len);
+          channel, target, args, creds, key, key_len TSRMLS_CC);
     } else {
       channel->wrapper = le->channel;
     }
@@ -511,6 +511,7 @@ GRPC_STARTUP_FUNCTION(channel) {
   INIT_CLASS_ENTRY(ce, "Grpc\\Channel", channel_methods);
   ce.create_object = create_wrapped_grpc_channel;
   grpc_ce_channel = zend_register_internal_class(&ce TSRMLS_CC);
+  gpr_mu_init(&global_persistent_list_mu);
   le_plink = zend_register_list_destructors_ex(
       NULL, php_grpc_channel_plink_dtor, "Persistent Channel", module_number);
   PHP_GRPC_INIT_HANDLER(wrapped_grpc_channel, channel_ce_handlers);

--- a/src/php/ext/grpc/channel_credentials.c
+++ b/src/php/ext/grpc/channel_credentials.c
@@ -165,7 +165,7 @@ PHP_METHOD(ChannelCredentials, createSsl) {
   }
 
   php_grpc_int hashkey_len = root_certs_length + cert_chain_length;
-  char hashkey[hashkey_len];
+  char *hashkey = emalloc(hashkey_len);
   if (root_certs_length > 0) {
     strcpy(hashkey, pem_root_certs);
   }
@@ -181,6 +181,7 @@ PHP_METHOD(ChannelCredentials, createSsl) {
       pem_key_cert_pair.private_key == NULL ? NULL : &pem_key_cert_pair, NULL);
   zval *creds_object = grpc_php_wrap_channel_credentials(creds, hashstr, false
                                                          TSRMLS_CC);
+  efree(hashkey);
   RETURN_DESTROY_ZVAL(creds_object);
 }
 

--- a/templates/package.xml.template
+++ b/templates/package.xml.template
@@ -24,8 +24,9 @@
    </stability>
    <license>BSD</license>
    <notes>
-  - Fixed hang bug when fork() was used #11814
-  - Channel are now by default persistent #11878
+  - Fixed a Windows installation issue #12108
+  - Fixed a MacOS mutex segfault #12109
+  - Fixed a ZTS compilation issue #12109
    </notes>
    <contents>
     <dir baseinstalldir="/" name="/">

